strapi/design-system
# feat : Add XS size variant to IconButton component

### What does it do?

This PR introduces an XS variant for the `IconButton` component, adding a smaller button size option with 4px padding (0.25rem) around the icon. The changes include:

- Added a new case for the `XS` size in the `IconButton` component to apply the appropriate padding (`padding-block: 0.25rem; - padding-inline: 0.25rem`).
- Updated the `BUTTON_SIZES` constant to include `XS` as a valid option: `export const BUTTON_SIZES = ['XS', 'S', 'M', 'L']`.
Adjusted documentation to reflect the addition of the new `XS` size.

### Why is it needed?

The addition of the `XS` variant for the `IconButton` component addresses the need for smaller buttons in certain parts of the CMS. This will allow for a more elegant UI and simplify the code by eliminating the need to manually define hover states and click areas for each instance. The XS size will provide a more compact and consistent option for icon-only buttons in small UI elements.

### How to test it?

To test the new XS size, follow these steps:

1. Navigate to the `IconButton` component in Storybook.
2. Select the `XS` size variant.
3. Verify that the button renders correctly with 4px padding around the icon and that it behaves as expected during interactions (hover, click).

Additionally, a video demonstrating the changes has been attached for visual reference, showing the component's behavior with the new XS size.

https://github.com/user-attachments/assets/84bb1b96-ce4d-466f-95e6-cacccd4cc24d




### Related issue(s)/PR(s)

#1820
